src/x509_scitokens_issuer/x509_scitokens_issuer.py

```diff
@@ -272,10 +276,21 @@ def token_issuer():
 
     creds = {}
     dn_cred = None
+    entry_num = 0
     for key, val in request.environ.items():
-        if key.startswith("GRST_CRED_AURI_"):
-            entry_num = int(key[15:]) # 15 = len("GRST_CRED_AURI_")
+        pattern = "GRST_CRED_AURI_"
+        if app.config.get("CMS", False):
+            pattern = "HTTP_CMS_AUTH"
+        if key.startswith(pattern):
+            if pattern == "HTTP_CMS_AUTH":
+                if key.endswith("_DN"):
+                    val = "dn:"+val
+                eif key.endswith("_LOGIN"):
+                    val = "username:"+val
+                elif key.startswith("HTTP_CMS_AUTHZ"):
+                    val = "fqan:/Role={}".format(val.split(':')[-1])
```

Comment:
Since we're abusing the notation a bit, we might as well make it simpler and set the following:

```suggestion
                    val = "fqan:/{}".format(val.split(':')[-1])
```

Reply:
This will not work since FQANMatcher expects /Role= or /Capability=. Here is the code:
https://github.com/scitokens/x509-scitokens-issuer/blob/master/src/x509_scitokens_issuer/x509_scitokens_issuer.py#L77-L83.
If I supply value `'fqan:/brian'` it will not find a role and therefore later in a code the scopes will be empty.

Reply:
That should be fine (or, at least, it's intended to be fine - bug if it's not).  The result of that function if fed `fqan:/brian` should be `('/brian', None)`.

If you look at the example rule I gave:

https://github.com/scitokens/x509-scitokens-issuer#mapping-rules-and-mapping-file

```
{
  "rules": [
    {"match": "fqan:/cms", "scopes": ["read:/store", "write:/store/user/{username}"]}
  ]
}
```

this should still work without roles.

Reply:
ok, fixed, I found that I had wrong syntax in rules.json (I added /Role= over there). So code is up-to-date. Please review and merge.